internal/pkg/parser/testcase.go

```diff
@@ -43,18 +43,23 @@ type TestCase struct {
 
 // Request define the crafted http request present in the test case file.
 type Request struct {
-	Authority []string          `yaml:"authority"`
-	Method    []string          `yaml:"method"`
-	URI       []string          `yaml:"uri"`
-	Headers   map[string]string `yaml:"headers"`
+	Authority []string `yaml:"authority"`
+	Method    []string `yaml:"method"`
+	URI       []string `yaml:"uri"`
+	// optional fields
+	Headers     map[string]string `yaml:"headers"`
+	QueryParams map[string]string `yaml:"queryParams"`
+	Scheme      string            `yaml:"scheme"`
 }
 
 // Input contains the data structure which will be used to assert
 type Input struct {
-	Authority string
-	Method    string
-	URI       string
-	Headers   map[string]string
+	Authority   string
+	Method      string
+	URI         string
+	Headers     map[string]string
+	Scheme      string
+	QueryParams map[string]string
```

Comment:
Do we need a separate field for the parameters or can we use the `uri` and parse from it?

Also, the same parameter can be set multiple times and interpreted differently  by web frameworks as there is no apparent standard, this would be another reason not to use map[string]string

https://go.dev/play/p/dcExJ0RlFYT

Reply:
Sounds reasonable - let's use `URI` by default in that case. The main reason was to have the test case input be more similar to the virtual service (in the sense of having the queryParams be their own node), but I think avoiding `map[string]string` makes sense.

Reply:
As discussed will keep as is for now - the envoy behaviour is to check the first query param